src/sentry/sentry_metrics/aggregation_option_registry.py

```diff
@@ -22,17 +23,41 @@ class TimeWindow(Enum):
     "d:transactions/measurements.lcp@millisecond": {AggregationOption.HIST: TimeWindow.NINETY_DAYS},
 }
 
-USE_CASE_AGG_OPTION = {UseCaseID.CUSTOM: {AggregationOption.TEN_SECOND: TimeWindow.SEVEN_DAYS}}
+# Currently there are no default per-use case aggregation overrides
+# They are all set via options
+USE_CASE_AGG_OPTION = {}
 
 
-def get_aggregation_options(mri: str) -> dict[AggregationOption, TimeWindow] | None:
+def set_use_case_aggregation_options():
+    # As of now this option is still restricted
+    # to the scope of the custom use case
+    if options.get("sentry-metrics.10s-granularity"):
+        USE_CASE_AGG_OPTION[UseCaseID.CUSTOM] = {
+            AggregationOption.TEN_SECOND: TimeWindow.SEVEN_DAYS
+        }
+
+    for use_case in options.get("sentry-metrics.drop-percentiles.per-use-case"):
+        USE_CASE_AGG_OPTION[UseCaseID(use_case)] = {
+            AggregationOption.DISABLE_PERCENTILES: TimeWindow.NINETY_DAYS
+        }
+
+
+def get_aggregation_options(mri: str, org_id: int) -> dict[AggregationOption, TimeWindow] | None:
+
+    # Set various aggregation options that
+    # are use case-wide aggregations
+    set_use_case_aggregation_options()
```

Comment:
This function internally modifies a global, but this is done on every call of `get_aggregation_options`. The global isn't really needed therefore, it could be just a local variable. This would further allow you to simplify and inline the checks into a single `if`-chain below that would also more clearly show precedence of the options.

Reply:
Updated